Grids toolbar: add description (#2626)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxTreeList/1 Configuration/toolbar.md

---
---
##### shortDescription

> Customizes toolbar.

Reviewer: ```suggestion
Configures the toolbar.
```

---

#include common-demobutton with {

>     url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeList/UICustomization/"

Reviewer: Is there gonna be a new demo targeting the TreeList specifically?

}

Toolbar allows you to configure predefined items, add custom items, and reorder them all.

> **Predefined Items**

Reviewer: ```suggestion
## Predefined Controls
```

> Predefined items include:

Reviewer: ```suggestion
Predefined controls appear on the toolbar depending on whether a specific DataGrid feature is enabled. The following table illustrates the dependency:
```

<table class="dx-table">
  <tr>

>     <th>Picture</th>
>     <th>Item</th>
>     <th>Prerequisites</th>

Reviewer: Let's reorder the table columns:

```suggestion
    <th>Control Name</th>
    <th>Image</th>
    <th>Prerequisites</th>
```

  </tr>
  <tr>
    <td><img src="/images/DataGrid/icons/toolbar_add.png" alt="DevExtreme HTML5 JavaScript jQuery Angular Knockout UI component TreeList Toolbar AddButton" style="margin:1px; vertical-align:middle"/></td>
    <td><i>addRowButton</i></td>

>     <td>editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding">allowAdding</a> is set to <b>true</b></td>

Reviewer: Let's add full links. Make `editing` a link too.

```suggestion
    <td>editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding">allowAdding</a> is <b>true</b></td>
```

    <td><i>revertButton</i></td>
    <td>editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode">mode</a> is set to <i>"batch"</i></td>
  </tr>
  <tr>
    <td><img src="/images/DataGrid/icons/toolbar_save.png" alt="DevExtreme HTML5 JavaScript jQuery Angular Knockout UI component TreeList Toolbar SaveButton" style="margin:1px; vertical-align:middle"/></td>
    <td><i>saveButton</i></td>
    <td>editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode">mode</a> is set to <i>"batch"</i> and editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating">allowUpdating</a> is set to <b>true</b></td>
  </tr>
  <tr>
    <td><img src="/images/DataGrid/icons/toolbar_searchPanel.png" alt="DevExtreme HTML5 JavaScript jQuery Angular Knockout UI component TreeList Toolbar searchPanel" style="margin:1px; vertical-align:middle"/></td>
    <td><i>searchPanel</i></td>
    <td>searchPanel.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/searchPanel/#visible">visible</a> is set to <b>true</b></td>
  </tr>
</table>

> To add a predefined item, add its string to the toolbar array. If you need to configure an item, add an object instead and specify the [name](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/#name) and other [properties](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/).

Reviewer: ```suggestion
If you need to customize a predefined item, add an object to the **items[]** array. This object must contain the item's [name]({currentpath}/#name) and [properties]() that you want to customize. If an item does not need customization, simply include its name in the **items[]** array.

[important] The TreeList does not display controls missing from the **items[]** array. Ensure that this array includes controls for all enabled features.
```

> **Custom Items**

Reviewer: ```suggestion
## Custom Controls
```

> Custom items can display any html code.  To add such an item, specify its [template](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/#template) and other properties. To embed a DevExtreme component, specify the [widget](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/#widget) and other [properties](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/) instead.
> 
> [note] If you use toolbar to specify a custom item and/or configure a predefined item, add corresponding strings for other used features. Otherwise, the TreeList will not display these controls.

Reviewer: Make the descriptions based on the [Toolbar Customization](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/ToolbarCustomization/jQuery/Light/) demo.

```suggestion
- DevExtreme components
...

- Custom elements
...
```
